lms/djangoapps/learner_home/test_serializers.py

```diff
@@ -751,6 +751,37 @@ def test_programs_serializer(self):
                 ).data
             },
         )
+    
+    def test_empty_source_programs_serializer(self):
+        """Test the ProgramsSerializer with empty data"""
+        # Given a program with empty test data
+        input_data = self.generate_test_related_program()
+
+        input_data["banner_image"] = None
+        input_data["title"] = None
+        input_data["type"] = None
+
+        # When I serialize the program
+        output_data = RelatedProgramSerializer(input_data).data
+
+        # Test the output
```

Comment:
Nit: you can probably simplify this test to just check the fields you care about.